From 20f19d19c4903a606d32e6313c7609f3436fe138 Mon Sep 17 00:00:00 2001
From: mayeut <mayeut@users.noreply.github.com>
Date: Sat, 20 Apr 2024 13:43:28 +0200
Subject: [PATCH] fix: use `xmlError cont*` instead of xmlErrorPtr

This fixes issues on Xcode15 (at least) when building against libxml 2.12+
---
 src/lib/src/tdf_xml_validator.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/lib/src/tdf_xml_validator.cpp b/src/lib/src/tdf_xml_validator.cpp
index bbf0390..57670e0 100644
--- a/src/lib/src/tdf_xml_validator.cpp
+++ b/src/lib/src/tdf_xml_validator.cpp
@@ -19,7 +19,7 @@ using namespace virtru;
 // Based on https://stackoverflow.com/questions/54124989/libxml2-get-xsd-validation-errors
 
 // This routine is a callback for the parser, used when there is an error validating the supplied XML against the schema
-static void schemaParseErrorHandler(void *arg, xmlErrorPtr error)
+static void schemaParseErrorHandler(void *arg, xmlError const* error)
 {
     std::ostringstream errorMsg;
 
@@ -95,7 +95,7 @@ bool TDFXMLValidator::validate(xmlTextReaderPtr reader) {
 
                 if (moreToRead != 0) {
                     // There was an error parsing the supplied XML
-                    xmlErrorPtr error = xmlGetLastError();
+                    xmlError const* error = xmlGetLastError();
                     std::ostringstream errorMsg;
                     errorMsg << "Schema validation error " << error->file << "(" << error->line << ") " << error->int2 << " " << error->code << " " << error->message;
                     ThrowException(errorMsg.str(), VIRTRU_TDF_FORMAT_ERROR);
@@ -107,4 +107,4 @@ bool TDFXMLValidator::validate(xmlTextReaderPtr reader) {
         }
     }
     return retval;
-}
\ No newline at end of file
+}
